src/txn/txn.c

```diff
@@ -1296,6 +1321,23 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
           upd->txnid == upd->next->txnid && upd->start_ts == upd->next->start_ts));
     first_committed_upd_in_hs =
       first_committed_upd != NULL && F_ISSET(first_committed_upd, WT_UPDATE_HS);
+
+    /* Set the flag to delete the value in history store. */
```

Comment:
This needs a better comment. we are marking the updates followed by a prepared update that is present in the history store to be removed.